### DISCUSSION TITLE
Public School App with upside potentials needed<title>

### DISCUSSION THREAD

**Discussion by kilogram6gram** [community/community#103048] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

App Purpose: To restrict access by students to social media accounts while in the presence of a predefined school WiFi signal.

I am putting this appeal here, knowing that the talent here exists to get such a project off the ground relatively quickly. I am a public school substitute teacher that has witnessed extreme problems affecting youth who can’t control their time on social media. Very few teachers will oppose this idea. 

The social media accounts to be restricted would be determined on an individual school district basis, or on an individual school basis. Schools currently require students to sign into Canvas, Blackboard, Google Classroom, or other sites to do much of their work.

Basically, the core of the app would be like a firewall, but on the portable device itself. It will need to work against WiFi access and regular cellular access. It also needs to have the ability to not be shut off simply because a student turns off the device's WiFi access, so it may need a dynamic calendaring component. How schools determine how the apps get installed will depend upon the app design, school district policy, and how the legal establishment responds to this technology. 

The app needs to be able to be used on all portable devices students may use, initially for cell phones, then extending to including other portable devices.

A future version could access a district database on the school district intranet. It would allow individual teachers to exclude restrictions against individual videos and media. It could also allow for time-limited access for students within a specific Active Directory (or other system grouping) grouping. This would require a web interface for individual teachers to access. 

Currently, school districts try to use firewalls to restrict gaming and social media websites, but this approach doesn’t keep the students from using phones to access those. Furthermore, URL redirection is often used by gaming websites to bypass school firewalls… and the students quickly find those. The web-access aspect by teachers can also aid school districts’ IT departments meet the ever-changing needs of the students. 

I am not interested in leading the project, because I don’t have the background to do that. It is my hope that you who have the skills and abilities will create applications that can meet the needs of the students, schools, and society.

**Comment by Gansunshine** [comment]
Creating an application with the described features involves a multifaceted approach, combining elements of network security, mobile device management, and educational technology. Here's a broad outline of how such an application could be developed:

1. **Define Requirements and Features:**
   - Clearly outline the features and requirements of the application based on your description. This includes specifying the types of social media to be restricted, integration with school databases, and the ability for teachers to customize restrictions.

2. **Technical Architecture:**
   - Determine the technical architecture of the application, considering compatibility with various devices and operating systems. Decide whether it will be a standalone app or require integration with existing school systems.

3. **Network Security:**
   - Develop a robust network security mechanism that can identify the school's WiFi signal and activate restrictions on social media apps. This may involve creating a local network proxy or VPN solution.

4. **Device Management:**
   - Implement device management features to ensure that the app cannot be easily disabled by students. This may include dynamic calendaring for automatic activation/deactivation of restrictions.

5. **User Authentication:**
   - Establish a secure user authentication system to ensure that only authorized individuals (teachers, administrators) can make changes or exceptions to the restrictions.

6. **Integration with School Systems:**
   - If accessing a district database, design a secure method for the app to communicate with the school's intranet. This could involve APIs, secure connections, and adherence to any relevant data protection regulations.

7. **Teacher Web Interface:**
   - Create a user-friendly web interface for teachers to customize restrictions, manage exceptions, and view reports. This may involve a role-based access control system.

8. **Testing:**
   - Thoroughly test the application in various school environments and on different devices to ensure its reliability and effectiveness.

9. **Privacy and Legal Considerations:**
   - Address privacy concerns and comply with legal regulations, especially when dealing with student data and access restrictions.

10. **Deployment and Support:**
    - Develop a deployment plan that considers how the app will be distributed to students' devices. Provide ongoing support and updates to address any issues or adapt to changes in technology.

11. **Community and Stakeholder Engagement:**
    - Involve teachers, administrators, and possibly parents in the development process to gather feedback and ensure the application meets the practical needs of the educational community.

Given the complexity and sensitivity of such an application, it would be beneficial to collaborate with educational technology experts, network security professionals, and legal advisors to ensure the success and ethical use of the tool. It's also essential to get approval and support from the relevant educational authorities and stakeholders before deploying such a solution in a school setting.
